electron/storage/AddressBookStorage.ts

```diff
@@ -78,6 +78,13 @@ class AddressBookStorage extends AbstractStorage<Contact> {
   }
 
   async add(entity: Omit<Contact, '_id' | 'order'>): Promise<Contact> {
+    const record = await this.find({ address: entity.address })
+    if (record) {
+      throw {
```

Comment:
If this is the way things have to be, let's merge it, but it seems a little weird to throw a non-error then convert it back into an error at a higher level -- It'd feel a bit better to not need the intermediate layer that converts objects to errors and re-throws them, or at least if the object were a subclassed error instead that we could specifically check for.

I think `message` exists on Error instances by default, so that case should get hit almost every time

Reply:
Yep this is my bad here. I pushed up some intermediate work. Fixed